lib/iris/fileformats/netcdf.py

```diff
@@ -780,6 +784,13 @@ def write(self, cube, local_keys=None, unlimited_dimensions=None,
                     'cf_patch')
                 warnings.warn(msg)
 
+        # Perform any user registered callback function.
+        # As the NetCDF saver commits to disk as it processes the cube and
+        # its metadata, it is not possible to skip or ignore the derived
+        # NetCDF representation of a cube as with other file format savers.
```

Comment:
> The scheme I've knocked together only has the original cube before it was sliced up.

I've addressed that issue by extending my demo PR to include deconstructed save for PP. (Which also re-writes the `save` function to use the new deconstructed pieces so we get a bunch of testing for free. :grin:)